### ISSUE TITLE
[Bug]: Missing quant_config in MTP eh_proj causes severe silent precision loss during W8A8 quantization

### ISSUE THREAD

**Issue by tubaii** [vllm-project/vllm#48492]
### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>

### 🐛 Describe the bug

When running models with Multi-Token Prediction (MTP) enabled under W8A8 (or similar) quantization, severe silent precision loss occurs. 
Through dump analysis comparing original (BF16) and W8A8 weights, we discovered that the `eh_proj` linear layer in multiple MTP implementations uses standard `nn.Linear` without passing the `quant_config`. This causes the quantized checkpoint weights to be incorrectly loaded and computed, resulting in massive numerical distortion and degraded generation quality.
### Root Cause Analysis
In several MTP model files, `eh_proj` is defined as a standard `nn.Linear`:
```python
# BUG: nn.Linear doesn't accept quant_config
self.eh_proj = nn.Linear(config.hidden_size * 2, config.hidden_size, bias=False)
```
When W8A8 quantization is enabled:
1. The checkpoint provides INT8 quantized weights (value range `[-127, 127]`) and corresponding per-channel scales for this layer.
2. Because `nn.Linear` lacks quantization awareness (no `quant_config`), the default weight loader directly loads the INT8 integers into a BF16 tensor without applying the dequantization scales.
3. During the forward pass, `F.linear` executes standard floating-point multiplication using these massive integer values, causing the output norm to amplify by ~1116x compared to the unquantized baseline.
**Why is this a "silent" precision loss?**
The immediate next layer after `eh_proj` is `input_layernorm` (RMSNorm). Mathematically, RMSNorm normalizes the macroscopic magnitude, which cancels out the ~1116x global amplification. This prevents an immediate crash (like NaNs), making the issue hard to spot casually. 
However, the precision is permanently destroyed because:
* **Loss of per-channel proportions:** W8A8 uses per-channel quantization. Bypassing the scale destroys the relative size relationships between feature channels. RMSNorm only fixes the global norm, not the internal channel distortions.
* **BF16 precision truncation:** The massive amplification causes large numbers to collide with BF16's precision limits ("large numbers eat small numbers"), permanently discarding subtle numerical differences before normalization.
### Proposed Fix
Replace `nn.Linear` with `ReplicatedLinear` and pass the `quant_config` so the weights are correctly dequantized during forward passes. `ReplicatedLinear` is appropriate here because the input to `eh_proj` is already TP-sharded from `cat(enorm, hnorm)`, and its output goes directly into the TP-aware `mtp_block`, meaning no cross-rank communication is needed.
```diff
- self.eh_proj = nn.Linear(config.hidden_size * 2, config.hidden_size, bias=False)
+ from vllm.model_executor.layers.linear import ReplicatedLinear
+ self.eh_proj = ReplicatedLinear(
+     config.hidden_size * 2,
+     config.hidden_size,
+     bias=False,
+     quant_config=quant_config,
+     params_dtype=config.dtype,
+ )
```
### Affected Files
This bug exists in at least 7 MTP model implementations:
- `vllm/model_executor/models/hy_v3_mtp.py` (~Line 100)
- `vllm/model_executor/models/deepseek_mtp.py` (~Line 73)
- `vllm/model_executor/models/glm4_moe_lite_mtp.py` (~Line 93)
- `vllm/model_executor/models/glm4_moe_mtp.py` (~Line 89)
- `vllm/model_executor/models/glm_ocr_mtp.py` (~Line 66)
- `vllm/model_executor/models/openpangu_mtp.py` (~Line 61)
- `vllm/model_executor/models/step3p5_mtp.py` (~Line 54)
*(Note: Implementations like `longcat_flash_mtp.py` and `mimo_v2_mtp.py` already correctly use `ReplicatedLinear` with `quant_config` and serve as good references.)*
### Additional context
Applying this fix ensures that MTP layers correctly handle quantized weights, restoring the intended precision and preventing subtle output quality degradation during W8A8 inference.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by ErenAta16** [comment]
@tubaii the dump-based root-cause analysis here is excellent, the RMSNorm-masking explanation especially. Opened #48506 applying the fix across all seven files.

One thing worth flagging: `ReplicatedLinear` returns a `(output, bias)` tuple, unlike `nn.Linear`. The diff sketched above only touches the layer definition, applying it as-is would leave `hidden_states = self.eh_proj(...)` trying to unpack a tuple into a single variable at every call site. The PR updates both the definition and the call site in each file to match the working pattern already in `longcat_flash_mtp.py`.
